File: Java/DesignPatterns/src/main/java/designPatterns/behavioral/state/ConcreteState.java
package designPatterns.behavioral.state;

public class ConcreteState {
  static class StartState implements State {
    @Override
    public String doAction(Context context) {
      context.setState(this);
      return "Player is in start state";
    }

    @Override
    public String toString() {
      return "StartState";
    }
  }

  static class StopState implements State {
    @Override
    public String doAction(Context context) {
      context.setState(this);
      return "Player is in stop state";
    }

    @Override
    public String toString() {
      return "StopState";
    }
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/behavioral/state/ConcreteState.kt
package behavioral.state

sealed class ConcreteState {
  class StartState : State {
    override fun doAction(context: Context): String {
      context.state = this
      return "Player is in start state"
    }

    override fun toString(): String = "StartState"
  }

  class StopState : State {
    override fun doAction(context: Context): String {
      context.state = this
      return "Player is in stop state"
    }

    override fun toString(): String = "StopState"
  }
}
